### PR TITLE
replxx : add version 0.0.4 and modernize recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **replxx/0.0.4**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/replxx/all/conanfile.py [hunk 1]
```diff
@@ -10,10 +11,10 @@ class ReplxxConan(ConanFile):
     """
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/AmokHuginnsson/replxx"
-    topics = ("conan", "readline", "libedit", "UTF-8")
+    topics = ("readline", "libedit", "UTF-8")
     license = "BSD-3-Clause"
     exports_sources = ["CMakeLists.txt"]
-    generators = "cmake", "cmake_find_package"
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
```
Comment (madebr) [inline recipes/replxx/all/conanfile.py:17]: Are the `cmake_find_package` and `cmake_find_package_multi` generators needed at all?
This package doesn't have any requirements.
```suggestion
    generators = "cmake"
```

Comment (toge) [reply recipes/replxx/all/conanfile.py:17]: You are right. Thank you.
I delete this.

Comment (toge) [reply recipes/replxx/all/conanfile.py]: Thank you.
I misunderstand the meaning of cmake_find_package.
I will fix it.

Comment (madebr) [reply recipes/replxx/all/conanfile.py]: No problem.
When the port to conan 2.0 happens, we will have to go through every recipe and replace `cmake` with `CMakeToolchain`/`CMakeDeps` anyway.

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**